spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/StreamService.java

```diff
@@ -51,7 +65,9 @@
 	 * @param streamName the name of the stream to update
 	 * @param updateStreamRequest the UpdateStreamRequest to use during the update
 	 */
-	void updateStream(String streamName, UpdateStreamRequest updateStreamRequest);
+	default void updateStream(String streamName, UpdateStreamRequest updateStreamRequest) {
+		throw new IncompatibleStreamDeployerException(StreamDeployers.appdeployer.toString());
```

Comment:
Do we really want to refer `appDeployer` here? Maybe it is better to keep this generic and refer `appDeployer` in its own StreamService.

Reply:
Ok, i will remove all defaults from the `StreamDeployer` and `StreamService` interfaces and move the exception throwing to the implementing classes.